tests/mgr/test_restful_operations_pool_endpoint.py

```diff
@@ -0,0 +1,134 @@
+import json
+
+from api.restful.pool import Pool
+from cli.cephadm.cephadm import CephAdm
+from cli.exceptions import OperationFailedError
+from utility.log import Log
+
+LOG = Log(__name__)
+
+
+# Request header
+HEADER = {
+    "Accept": "application/vnd.ceph.api.v1.0+json",
+    "Content-Type": "application/json",
+}
+
+# Missing config parameter err msg
+CONFIG_ERR_MSG = "Mandatory config parameter '{}' not provided"
+
+
+# Mandatory test config exception
+class ConfigNotFoundError(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    config = kw.get("config", {})
+
+    # Get the MGR node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+    mgr_ip = node.ip_address
+
+    # Read the test configs
+    username = config.get("username")
+    if not username:
+        raise ConfigNotFoundError(CONFIG_ERR_MSG.format("username"))
+
+    endpoint = config.get("endpoint")
+    if not endpoint:
+        raise ConfigNotFoundError(CONFIG_ERR_MSG.format("endpoint"))
+
+    method = config.get("method")
+    if not method:
+        raise ConfigNotFoundError(CONFIG_ERR_MSG.format("endpoint"))
+
+    status_code = config.get("status_code")
+    if not status_code:
+        raise ConfigNotFoundError(CONFIG_ERR_MSG.format("status_code"))
+
+    # Get key for API user
+    key = json.loads(CephAdm(node).ceph.restful.list_key()).get(username, None)
+    if not key:
+        raise OperationFailedError(f"Failed to get key for API user {username}")
+
+    # Declare the URL for the request
+    url = f"https://{mgr_ip}:8003"
+
+    if method == "GET":
```

Comment:
better to create separate methods for `get`, `post` etc methods.

Reply:
Done